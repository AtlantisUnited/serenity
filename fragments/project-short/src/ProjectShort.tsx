import React                                    from 'react'

import { PersonIcon, VisibilityIcon, WorkIcon } from '@ui/icons'
import { Label }                                from '@ui/label'
import { Box, Column, Layout, Row }             from '@ui/layout'
import { NextLink as Link }                     from '@ui/link'
import { RelativeTime }                         from '@ui/relative-time'
import { Text }                                 from '@ui/text'

import messages                                 from './messages'

export const ProjectShort = ({
  intl,
  id = '',
  name = '',
  status = '',
  budget = '',
  description = '',
  category = {
    name: '',
  },
  replyStatus = '',
  legalEntitiesOnly = false,
  replyCount = 0,
  views = 0,
  type = 'projects',
  short = false,
  publicationDate,
}: any) => (
  <Column>
    <Layout>
      <Row justifyContent='space-between'>
        <Layout>
          {id && (
            <Link href={`/${type}/${id}`}>
              <Text
                fontWeight='medium'
                fontSize={['extra', 'large']}
                lineHeight='extra'
                color='black'
              >
                {name}
              </Text>
            </Link>
          )}
          {!id && (
            <Text
              fontWeight='medium'
              fontSize={['extra', 'large']}
              lineHeight='extra'
              color='black'
            >
              {name}
            </Text>
          )}
        </Layout>
        <Layout flexBasis={20} />
        <Layout alignItems='center'>
          <Text
            fontWeight='medium'
            fontSize='extra'
            lineHeight='extra'
            color='mountainmist'
            whiteSpace='nowrap'
          >
            {intl.formatNumber(budget, {
              style: 'currency',
              currency: 'RUB',
              minimumSignificantDigits: 1,
            })}
          </Text>
        </Layout>
      </Row>
    </Layout>
    <Layout flexBasis='5px' />
    <Layout>
      <Text lineHeight='medium'>{description}</Text>
    </Layout>
    <Layout flexBasis={15} />
    <Layout>
      <Row alignItems={['flex-start', 'flex-start', 'flex-start', 'center']}>
        <Box flexDirection={['column-reverse', 'column-reverse', 'row']} flexGrow={1}>
          <Layout>
            <Text fontWeight='medium' lineHeight='medium' color='silver' px='6px' bg='alto'>
              {category.name}
            </Text>
          </Layout>
          <Layout flexBasis={16} />
          <Layout alignItems='center'>
            {messages[status] && <Label size='small'>{intl.formatMessage(messages[status])}</Label>}
          </Layout>
          <Layout flexGrow={1} />
          {replyStatus && (
            <>
              <Layout flexBasis={[10, 10, 40]} />
              <Layout alignItems='center'>
                <Text fontWeight='semiBold' fontSize='tiny' color='mountainmist'>
                  {intl.formatMessage(
                    (status === 'published' && messages.notSelected) ||
                      ((replyStatus === 'chosen' ||
                        replyStatus === 'performed' ||
                        replyStatus === 'completed') &&
                        messages.selected) ||
                      messages.notYouSelected
                  )}
                </Text>
              </Layout>
            </>
          )}
          {publicationDate && (
            <>
              <Layout flexBasis={[10, 10, 40]} />
              <Layout alignItems='center'>
                <Text fontSize='tiny' color='mountainmist'>
                  <RelativeTime value={publicationDate} />
                </Text>
              </Layout>
            </>
          )}
        </Box>
        <Layout flexBasis={[10, 20, 40]} />
        <Box flexDirection={['column-reverse', 'column-reverse', 'column-reverse', 'row']}>
          {legalEntitiesOnly && (
            <Layout pr={[0, 0, 0, 32]}>
              <WorkIcon width={16} height={16} />
            </Layout>
          )}
          <Layout flexBasis={[10, 10, 10, 0]} />
          <Layout pr={[0, 0, 0, 24]}>
            <PersonIcon width={16} height={16} />
            <Text
              fontWeight='semiBold'
              fontSize='tiny'
              color='mountainmist'
              pl={['4px', '4px', '8px']}
              whiteSpace='nowrap'
            >
              {replyCount}
              {short
                ? intl.formatMessage(messages.replyCount, {
                    сount: replyCount,
                  })
                : ''}
            </Text>
          </Layout>
          <Layout flexBasis={[10, 10, 10, 0]} />
          <Layout>
            <VisibilityIcon width={16} height={16} />
            <Text
              fontWeight='semiBold'
              fontSize='tiny'
              color='mountainmist'
              pl={['4px', '4px', '8px']}
              whiteSpace='nowrap'
            >
              {views}
              {short ? intl.formatMessage(messages.views, { сount: views }) : ''}
            </Text>
          </Layout>
        </Box>
      </Row>
    </Layout>
  </Column>
)
