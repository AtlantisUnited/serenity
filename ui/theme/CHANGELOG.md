# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.5 (2021-05-10)

**Note:** Version bump only for package @ui/theme

## 0.0.4 (2020-09-28)

### Bug Fixes

- **accounts:** add emotion-server, delete font ([08acf5a](https://github.com/Atlantis-Lab/serenity/commit/08acf5a7169352a606925fdfbbaa18f9917a8106))

## 0.0.3 (2020-06-30)

**Note:** Version bump only for package @ui/theme

## 0.0.2 (2020-06-10)

**Note:** Version bump only for package @ui/theme
