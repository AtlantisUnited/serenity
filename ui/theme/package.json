{
  "name": "@ui/theme",
  "version": "0.0.5",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@types/react": "^17.0.6",
    "@types/styled-system": "5.1.4"
  },
  "dependencies": {
    "@emotion/css": "^11.1.3",
    "@emotion/react": "^11.4.0",
    "@emotion/styled": "^11.3.0",
    "react": "^17.0.2",
    "styled-system": "^5.1.0"
  }
}
