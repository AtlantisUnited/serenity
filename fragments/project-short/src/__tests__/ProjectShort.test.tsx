/**
 * @jest-environment jsdom
 */

import { render }        from '@testing-library/react'

import React             from 'react'
import { IntlProvider }  from 'react-intl'
import { useIntl }       from 'react-intl'

import { ThemeProvider } from '@ui/theme'

import { ProjectShort }  from '../ProjectShort'

const TestComponent = () => {
  const intl = useIntl()

  return <ProjectShort intl={intl} />
}

describe('Customer projects list', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(
        <IntlProvider locale='ru' defaultLocale='ru' messages={{}}>
          <ThemeProvider>
            <TestComponent />
          </ThemeProvider>
        </IntlProvider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
