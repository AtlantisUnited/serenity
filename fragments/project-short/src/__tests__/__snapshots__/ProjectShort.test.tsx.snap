// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer projects list snapshots should match latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-27 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  color: rgb(0,0,0);
  font-weight: 500;
  font-size: 18px;
  line-height: 1.3;
  font-family: Inter;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 22px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 20px;
  -ms-flex-preferred-size: 20px;
  flex-basis: 20px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  white-space: nowrap;
  color: rgb(153,153,153);
  font-weight: 500;
  font-size: 18px;
  line-height: 1.3;
  font-family: Inter;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 5px;
  -ms-flex-preferred-size: 5px;
  flex-basis: 5px;
}

.emotion-8 {
  line-height: 1.6;
  font-family: Inter;
  font-weight: 400;
  font-size: 13px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 15px;
  -ms-flex-preferred-size: 15px;
  flex-basis: 15px;
}

.emotion-25 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media screen and (min-width:40em) {
  .emotion-25 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media screen and (min-width:52em) {
  .emotion-25 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media screen and (min-width:64em) {
  .emotion-25 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-16 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

@media screen and (min-width:40em) {
  .emotion-16 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media screen and (min-width:52em) {
  .emotion-16 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-11 {
  color: rgb(190,190,190);
  background-color: rgb(216,216,216);
  padding-left: 6px;
  padding-right: 6px;
  font-weight: 500;
  line-height: 1.6;
  font-family: Inter;
  font-size: 13px;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 16px;
  -ms-flex-preferred-size: 16px;
  flex-basis: 16px;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 10px;
  -ms-flex-preferred-size: 10px;
  flex-basis: 10px;
}

@media screen and (min-width:40em) {
  .emotion-17 {
    -webkit-flex-basis: 20px;
    -ms-flex-preferred-size: 20px;
    flex-basis: 20px;
  }
}

@media screen and (min-width:52em) {
  .emotion-17 {
    -webkit-flex-basis: 40px;
    -ms-flex-preferred-size: 40px;
    flex-basis: 40px;
  }
}

.emotion-24 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

@media screen and (min-width:40em) {
  .emotion-24 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media screen and (min-width:52em) {
  .emotion-24 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media screen and (min-width:64em) {
  .emotion-24 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 10px;
  -ms-flex-preferred-size: 10px;
  flex-basis: 10px;
}

@media screen and (min-width:40em) {
  .emotion-18 {
    -webkit-flex-basis: 10px;
    -ms-flex-preferred-size: 10px;
    flex-basis: 10px;
  }
}

@media screen and (min-width:52em) {
  .emotion-18 {
    -webkit-flex-basis: 10px;
    -ms-flex-preferred-size: 10px;
    flex-basis: 10px;
  }
}

@media screen and (min-width:64em) {
  .emotion-18 {
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
  }
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 0;
}

@media screen and (min-width:40em) {
  .emotion-20 {
    padding-right: 0;
  }
}

@media screen and (min-width:52em) {
  .emotion-20 {
    padding-right: 0;
  }
}

@media screen and (min-width:64em) {
  .emotion-20 {
    padding-right: 24px;
  }
}

.emotion-19 {
  white-space: nowrap;
  color: rgb(153,153,153);
  padding-left: 4px;
  font-weight: 600;
  font-size: 12px;
  font-family: Inter;
}

@media screen and (min-width:40em) {
  .emotion-19 {
    padding-left: 4px;
  }
}

@media screen and (min-width:52em) {
  .emotion-19 {
    padding-left: 8px;
  }
}

<div
    class="emotion-27"
    display="flex"
    width="100%"
  >
    <div
      class="emotion-1"
      display="flex"
    >
      <div
        class="emotion-5"
        display="flex"
        width="100%"
      >
        <div
          class="emotion-1"
          display="flex"
        >
          <span
            class="emotion-0"
            color="black"
            font-family="primary"
            font-size="extra,large"
            font-weight="medium"
          />
        </div>
        <div
          class="emotion-2"
          display="flex"
        />
        <div
          class="emotion-4"
          display="flex"
        >
          <span
            class="emotion-3"
            color="mountainmist"
            font-family="primary"
            font-size="extra"
            font-weight="medium"
          >
            0 ₽
          </span>
        </div>
      </div>
    </div>
    <div
      class="emotion-7"
      display="flex"
    />
    <div
      class="emotion-1"
      display="flex"
    >
      <span
        class="emotion-8"
        font-family="primary"
        font-size="normal"
        font-weight="normal"
      />
    </div>
    <div
      class="emotion-10"
      display="flex"
    />
    <div
      class="emotion-1"
      display="flex"
    >
      <div
        class="emotion-25"
        display="flex"
        width="100%"
      >
        <div
          class="emotion-16"
          display="flex"
        >
          <div
            class="emotion-1"
            display="flex"
          >
            <span
              class="emotion-11"
              color="silver"
              font-family="primary"
              font-size="normal"
              font-weight="medium"
            />
          </div>
          <div
            class="emotion-13"
            display="flex"
          />
          <div
            class="emotion-4"
            display="flex"
          />
          <div
            class="emotion-15"
            display="flex"
          />
        </div>
        <div
          class="emotion-17"
          display="flex"
        />
        <div
          class="emotion-24"
          display="flex"
        >
          <div
            class="emotion-18"
            display="flex"
          />
          <div
            class="emotion-20"
            display="flex"
          >
            <svg
              height="16"
              viewBox="0 0 12 12"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
                transform="translate(-2.000000, -2.000000)"
              >
                <path
                  d="M8,8 C9.47333333,8 10.6666667,6.80666667 10.6666667,5.33333333 C10.6666667,3.86 9.47333333,2.66666667 8,2.66666667 C6.52666667,2.66666667 5.33333333,3.86 5.33333333,5.33333333 C5.33333333,6.80666667 6.52666667,8 8,8 Z M8,9.33333333 C6.22,9.33333333 2.66666667,10.2266667 2.66666667,12 L2.66666667,13.3333333 L13.3333333,13.3333333 L13.3333333,12 C13.3333333,10.2266667 9.78,9.33333333 8,9.33333333 Z"
                  fill="#999999"
                  fill-rule="nonzero"
                  id="Shape"
                />
                <polygon
                  points="0 0 16 0 16 16 0 16"
                />
              </g>
            </svg>
            <span
              class="emotion-19"
              color="mountainmist"
              font-family="primary"
              font-size="tiny"
              font-weight="semiBold"
            >
              0
            </span>
          </div>
          <div
            class="emotion-18"
            display="flex"
          />
          <div
            class="emotion-1"
            display="flex"
          >
            <svg
              height="16"
              viewBox="0 0 16 10"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
                transform="translate(0.000000, -3.000000)"
              >
                <polygon
                  points="0 0 16 0 16 16 0 16"
                />
                <path
                  d="M8,3 C4.66666667,3 1.82,5.07333333 0.666666667,8 C1.82,10.9266667 4.66666667,13 8,13 C11.3333333,13 14.18,10.9266667 15.3333333,8 C14.18,5.07333333 11.3333333,3 8,3 Z M8,11.3333333 C6.16,11.3333333 4.66666667,9.84 4.66666667,8 C4.66666667,6.16 6.16,4.66666667 8,4.66666667 C9.84,4.66666667 11.3333333,6.16 11.3333333,8 C11.3333333,9.84 9.84,11.3333333 8,11.3333333 Z M8,6 C6.89333333,6 6,6.89333333 6,8 C6,9.10666667 6.89333333,10 8,10 C9.10666667,10 10,9.10666667 10,8 C10,6.89333333 9.10666667,6 8,6 Z"
                  fill="#999999"
                  fill-rule="nonzero"
                />
              </g>
            </svg>
            <span
              class="emotion-19"
              color="mountainmist"
              font-family="primary"
              font-size="tiny"
              font-weight="semiBold"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
